presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -450,6 +450,7 @@ public QueryInfo createQuery(SessionSupplier sessionSupplier, String query)
             }
         });
 
+        stats.queryQueued();
```

Comment:
we can move this line to `addStatsListener()` so that all stats related stuff is together in the same method (you can also add a comment saying that `QUEUED` is the initial state so we can immediately increment it etc.)? 
While you are at it we can also rename that method to `addStatsListeners`.